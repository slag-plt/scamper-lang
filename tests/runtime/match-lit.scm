(match 5
  [1 "fail"]
  [5 "numbers"])

(match "baz"
  ["foo" "fail"]
  ["bar" "fail"]
  ["baz" "strings"]
  ["boop" "fail"]) 

(match #\q
  [#\a "fail"]
  [#\q "chars"]
  [#\z "fail"])

(match #t
  [#f "fail"]
  [#t "bools"])

(match null
  [null "null"])

(match (list "lists" "a" "b")
  [null "fail"]
  [(cons head _) head])