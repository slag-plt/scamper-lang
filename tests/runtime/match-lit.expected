"numbers"
"strings"
"chars"
"bools"
"null"
"lists"
